src/oop/special_class_method/hard/q7/MainTest.java

```diff
@@ -0,0 +1,339 @@
+        Class<?>[] classArguments = {double.class, int.class};
+        assertTrue(testClass.hasConstructor(classArguments),
+                "Your " + className + " constructor does not have the correct parameters.");
+        assertTrue(testClass.hasModifier(classArguments, "public"),
+                "Your " + className + " constructor does not have the correct modifier.");
+    }
+
+    @Test
+    public void piggyBankClassHasRequiredConstructor2() {
+        Class<?>[] classArguments = {};
+        assertTrue(testClass.hasConstructor(classArguments),
+                "Your " + className + " constructor does not have the correct parameters.");
+        assertTrue(testClass.hasModifier(classArguments, "public"),
+                "Your " + className + " constructor does not have the correct modifier.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    @Tag("dependent1")
+    public void piggyBankClassHasCorrectConstructor(double value1, int value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, double.class},
+                {value2, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        String incorrectDoubleValueMessage =
+                "Your " + className + " constructor does not initialize the " + doubleFieldName + " attribute to the correct value.";
+        String incorrectStringValueMessage =
+                "Your " + className + " constructor does not initialize the " + intFieldName + " attribute to the correct value.";
+        _assertEquals(value1, testClass.getFieldValue(classInstance, doubleFieldName),
+                incorrectDoubleValueMessage);
+        _assertEquals(value2, testClass.getFieldValue(classInstance, intFieldName),
+                incorrectStringValueMessage);
+    }
+
+    @Test
+    public void piggyBankClassHasCorrectConstructor2() throws Throwable {
+        Object classInstance = testClass.createInstance();
+        String incorrectDoubleValueMessage =
+                "Your " + className + " constructor does not initialize the " + doubleFieldName + " attribute to the correct value.";
+        String incorrectStringValueMessage =
+                "Your " + className + " constructor does not initialize the " + intFieldName + " attribute to the correct value.";
+        _assertEquals(0.0, testClass.getFieldValue(classInstance, doubleFieldName),
+                incorrectDoubleValueMessage);
+        assertEquals(0, testClass.getFieldValue(classInstance, intFieldName),
+                incorrectStringValueMessage);
+    }
+
+    private static Stream<Arguments> doubleInputProvider() {
+        return Stream.of(
+                Arguments.of(4535.4, 432.2),
+                Arguments.of(234.4, 655.3),
+                Arguments.of(654.45, 453.3),
+                Arguments.of(43.3, 654.2)
+
+        );
+    }
+
+    @Test
+    public void setMoneyIsDefinedCorrectly() {
```

Comment:
I don't believe this and all the following "isDefinedCorrectly" methods are necessary